models/qwen3/14b/decode_layer.py

```
@@ -741,115 +709,135 @@ def _decode_layer(  # noqa: PLR0913 — model signature is intrinsic
-                for sb in pl.pipeline(1, os_ctx_blocks, stage=2):
-                    rec = os_g_base + sb * Q_HEAD_PAD
-                    # Partial load: tile stays Q_HEAD_PAD (16) rows, GM transfer = the 5
-                    # real rows (valid_shape on the slice → tile.load valid_shapes).
-                    oi_tmp_valid = pl.slice(
-                        all_oi_tmp, [Q_HEAD_PAD, HEAD_DIM], [rec, 0], valid_shape=[Q_HEAD_BATCH, HEAD_DIM]
+            # Phase-1 attn as a DATA-PARALLEL dual-AIV region: pl.split_aiv UP_DOWN
+            # halves the 16-row softmax (Q_HEAD_PAD) across the two AIV lanes (8 rows
+            # each); the compiler inserts aiv_shard at the QK C->V boundary and
+            # aic_gather at the exp->SV V->C boundary. aiv_id is unused (the halving is
+            # automatic via the region's own subblock index).
+            for aiv_id in pl.split_aiv(2, mode=pl.SplitMode.UP_DOWN):
```

Ruff (0.15.20)

[warning] Loop control variable `aiv_id` not used within loop body

(B007)